fcrepo-persistence-ocfl/src/main/java/org/fcrepo/persistence/ocfl/impl/DbFedoraToOcflObjectIndex.java

```diff
@@ -265,7 +267,11 @@ private void upsert(final String transactionId, final FedoraId fedoraId, final S
         parameterSource.addValue("ocflId", ocflId);
         parameterSource.addValue("transactionId", transactionId);
         parameterSource.addValue("operation", operation);
-        jdbcTemplate.update(UPSERT_MAPPING_TX_MAP.get(dbPlatform), parameterSource);
+        try {
+            jdbcTemplate.update(UPSERT_MAPPING_TX_MAP.get(dbPlatform), parameterSource);
+        } catch (final DataIntegrityViolationException e) {
```

Comment:
MariaDB is dying because it is throwing a different Exception and so its not getting caught here. 

```
org.springframework.jdbc.BadSqlGrammarException: PreparedStatementCallback; bad SQL grammar [INSERT INTO ocfl_id_map_session_operations (fedora_id, fedora_root_id, ocfl_id, session_id, operation) VALUES (?, ?, ?, ?, ?) ON DUPLICATE KEY UPDATE fedora_root_id = VALUES(fedora_id), ocfl_id = VALUES(ocfl_id), operation = VALUES(operation)]; nested exception is java.sql.SQLSyntaxErrorException: (conn=33) Data too long for column 'fedora_id' at row 1
```

So you'll need to catch both Exceptions here, ala 
```java
} catch (final DataIntegrityViolationException | BadSqlGrammarException e) {
```